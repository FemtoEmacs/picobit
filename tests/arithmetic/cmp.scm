(define-macro (test-cond condition name) ;; TODO watch out for name capture
  `(let ((DUMMY-x ,condition))
     (display ,name)
     (if DUMMY-x
       (display " : PASSED\n")
       (begin (display " : FAILED, got : ")
	      (display DUMMY-x)
	      (display "\n")))))

(test-cond (not (< 0 0)) "0 not neg")
(test-cond (< -1 0) "-1 neg")
(test-cond (= 5 5) "5 = 5")
(test-cond (< 2 5) "2 < 5")
(test-cond (> 5 2) "5 > 2")
(test-cond (= -5 -5) "-5 = -5")
(test-cond (> -2 -5) "-2 > -5")
(test-cond (< -5 -2) "-5 < -2")
(test-cond (< -5 65533) "-5 < 65533") ;; TODO still 16 bits, does the compiler encode it ok ? what about numbers over 16 bits long ?
(test-cond (< -5 2) "-5 < 2")
(test-cond (> 5 -65533) "5 > -65533")
(test-cond (> 5 -2) "5 > -2")
