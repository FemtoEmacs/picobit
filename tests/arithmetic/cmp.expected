:0080000080
:10800000FBD7880281684000812B4000813440000A
:108010000003FFFB0003FFFE815A40008165400022
:10802000816C4000816E400081714000817340008E
:10803000817840000004FFFD817C40008180400089
:108040008185400081864000813A400081424000A5
:10805000814F400081314000000300038123400034
:10806000811E400080240152804201538027015428
:108070008039000280320155803901568045015710
:1080800080390158804501598069015B8024015C79
:108090008066015D8040011C8073015E8069015F24
:1080A0008024016080390161804001628024016385
:1080B0008040016480480166803101678048016822
:1080C000804001698024016A8024016B8024016D55
:1080D0008024016F8036000280310170803101728E
:1080E00080400174805001758041014C80570176B9
:1080F0008039017780780179807001798031017A47
:108100008024017B8024017D8037017E8066017F91
:10811000807D018180390182802401838024018453
:10812000802401878024011F807800028024018838
:10813000803901898024018A8073018B8031013864
:1081400080240139806A0002802400028024013BDF
:10815000804D013C8024013D8057013E8031013F2C
:10816000806701408042016980730141807801434A
:10817000806B0002803E01448039014580730146D6
:108180008041014E807701478031014880300149AC
:108190008035014A800E000280240138803E014B68
:1081A0008040014C8027014D8042014E80390150B2
:1081B00080720151803A012C8027012D8039012ED7
:1081C0008027012F802401708049013080490132CD
:1081D000802401338036013580420002803601362A
:1081E000803E0155803401378054012480370002DD
:1081F000806E0125803901268071012780240128A5
:10820000806B01298072012A8024012080310121A4
:10821000804A01228031011F8024011D8042011CFF
:108220008078014A02B404DF5002B404D951040436
:10823000C8D402B404BB61EE0304C802B4049D6157
:10824000EE0909C702B4047F61EE0609C802B4044E
:108250006161EE0906CA02B4044361EEA107A107F9
:10826000C702B4042561EEA108A107CA02B404073D
:1082700061EEA107A108C802B403E961EEA107A15C
:1082800010C802B403CB61EEA10706C802B403AD67
:1082900061EE09A119CA02B4038F61EE09A108CAEF
:1082A00002B402A761EED801A10CB589EE20B30398
:1082B0007FA105B6800120DAB8954121DBB68002A6
:1082C00021CC9A21CE2161EE2022CFB67200EF2080
:1082D000B6800120DA9EA02607E620B558EEA0263B
:1082E00007E600EF20C0B8BF20B582B648012004E1
:1082F000C8B8AEA03121B5A402B583CDDCEF02A091
:1083000034210EC5C122CD2102CDB4030E7101214D
:108310000EC89220EF210E406221B662010421C2F4
:10832000EF2002B559B75520CCB8B8A02C07E620ED
:10833000CEB51FEE20CFB6800120D296A02D07E645
:1083400000EF20CC9DA02407E620CEB505EE20CF7F
:10835000B666A113B002B5EE20B002D2EEA02D0792
:10836000E600EF20D995A10EB102B520F09C209532
:10837000A10BB102B5A11BB102B5A10DB102B5A10E
:1083800006B002B5EE20B002B5EEA104B102B5010F
:10839000A117B002B5EE2095A105B102B5A106B0B6
:1083A00002B5EE20B002B5EEA104B102B501A109FB
:1083B000B002B5EE2095A105B102B5A106B002B597
:1083C000EE20B002B5EEA104B102B501A114B002D5
:1083D000B5EE2095A105B102B5A106B002B5EE201B
:1083E000B002B5EEA104B102B501A10AB002B5EE2A
:1083F0002095A105B102B5A106B002B5EE20B002EC
:10840000B5EEA104B102B501A116B002B5EE2095FA
:10841000A105B102B5A106B002B5EE20B002B5EEDD
:10842000A104B102B501A115B002B5EE2095A105D8
:10843000B102B5A106B002B5EE20B002B5EEA104BE
:10844000B102B501A112B002B5EE2095A105B102AD
:10845000B5A106B002B5EE20B002B5EEA104B1029E
:10846000B501A10FB002B5EE2095A105B102B5A1ED
:1084700006B002B5EE20B002B5EEA104B102B5011E
:10848000A11AB002B5EE2095A105B102B5A106B0C2
:1084900002B5EE20B002B5EEA104B102B501A118FB
:1084A000B002B5EE2095A105B102B5A106B002B5A6
:1084B000EE20B002B5EEA104B102B501A111B002E7
:1084C000B5EE2095A105B102B5A106B002B5EE202A
:1084D000B002B5EEA104B102B5012007E600EF023B
:1084E0002004C8212302CDCDB404EC71012204C8BC
:1084F00022242302CDCDCDB404FB710123B8DC21AD
:10850000B0031C23222602CDCDCDB4050E7101226D
:10851000B8C621B0031C21242602CDCDCDB405213F
:1085200071012022242602CDCDCDCDB4052F7000BF
:108530002121C4232502CDCDB4053C710122B8937D
:10854000219220EF2298019420B1031C20EF21B842
:0E8550007BB77500B86EB76A21B73B21B7251F
:00000001FF
